src/corelibs/U2Formats/src/BAMUtils.cpp

```diff
@@ -606,27 +631,40 @@ static int bamMergeCore(const QString& outFileName, const QList<QString>& filesT
     return 0;
 }
 
-GUrl BAMUtils::mergeBam(const QStringList& bamUrls, const QString& mergedBamTargetUrl, U2OpStatus& os) {
-    coreLog.details(tr(R"(Merging BAM files: "%1". Resulting merged file is: "%2")")
-                        .arg(QString(bamUrls.join(",")))
-                        .arg(QString(mergedBamTargetUrl)));
+static constexpr int MAX_FILES_OPENED = 100;
+static int recursiveBamMergeCore(const QString& outFileName, const QList<QString>& filesToMerge) {
+    int size = filesToMerge.size();
+    CHECK(size != 0, -1);
 
-    int rc = bamMergeCore(mergedBamTargetUrl, bamUrls);
-    CHECK_EXT(rc >= 0, os.setError(tr("Failed to merge BAM files: %1 into %2").arg(bamUrls.join(",")).arg(mergedBamTargetUrl)), {});
-    return mergedBamTargetUrl;
-}
+    int remainder = size % MAX_FILES_OPENED;
+    int iterationNumber = size / MAX_FILES_OPENED + (remainder != 0 ? 1 : 0);
+    if (iterationNumber == 1) {
+        return localBamMergeCore(outFileName, filesToMerge);
+    }
 
-void* BAMUtils::loadIndex(const QString& filePath) {
-    // See bam_index_load_local.
-    QString mode = "rb";
-    NP<FILE> file = BAMUtils::openFile(filePath + ".bai", mode);
-    if (file == nullptr && filePath.endsWith("bam")) {
-        file = BAMUtils::openFile(filePath.chopped(4) + ".bai", mode);
+    QStringList newOutFileNameList;
+    for (int i = 0; i < iterationNumber; i++) {
```

Comment:
A big part of the method here is a boilerplate that deals with chunks of files of size `MAX_FILES_OPENED`.
We already have a common pattern to deal with chunks with a minimal boilerplate. 
Could you please the manual chunk/remainder/iteration to `U2Region::split()` ?